pkg/resources/virtual/common/common.go

```diff
@@ -38,8 +38,12 @@ func (d *DefaultFields) transform(obj any) (any, error) {
 	if err != nil {
 		return nil, err
 	}
+	return d.transformCommon(raw)
+}
+
+func (d *DefaultFields) transformCommon(raw *unstructured.Unstructured) (any, error) {
```

Comment:
If we intend this function to be re-used by other transform functions, then it should be exported (rather then re-used by other functions in the same package).

I would recommend looking more into the interface here - since an object could go through multiple transform functions, I do like that we are avoiding the marshal to unstructured and the signal error checking, but I don't like that we are doing so through embedded objects and use of private methods.